elasticgraph-warehouse/lib/elastic_graph/warehouse/schema_definition/schema_artifact_manager_extension.rb

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      # Extension module for {ElasticGraph::SchemaDefinition::SchemaArtifactManager} that adds
+      # warehouse artifact generation support.
+      #
+      # @private
+      module SchemaArtifactManagerExtension
+        # Adds warehouse artifact to the artifact list after initialization.
+        #
+        # This method is called automatically after the SchemaArtifactManager is initialized.
+        # It checks if the schema has warehouse table definitions and adds the data_warehouse.yaml
+        # artifact if any tables are defined.
+        #
+        # @return [void]
+        def add_warehouse_artifact
+          # Only add the artifact if the results support warehouse_config (i.e., have the extension)
+          # :nocov: -- Defensive check; in practice when FactoryExtension is used, results will always have warehouse_config
+          return unless schema_definition_results.respond_to?(:warehouse_config)
+          # :nocov:
+
+          warehouse_config = schema_definition_results.warehouse_config
+
+          # Only add the artifact if there are warehouse tables defined.
+          return if warehouse_config.empty?
+
+          warehouse_artifact = ElasticGraph::SchemaDefinition::SchemaArtifact.new(
+            ::File.join(@schema_artifacts_directory, ::ElasticGraph::Warehouse::DATA_WAREHOUSE_FILE),
```

Comment:
```suggestion
            ::File.join(@schema_artifacts_directory, DATA_WAREHOUSE_FILE),
```

The `module ElasticGraph; module Warehouse` lines above allow this constant to resolve without needing to fully qualify it.